### ISSUE TITLE
[uart] V1 Signoff

### ISSUE THREAD

**Issue by msfschaffner** [lowRISC/opentitan#21019]
### Description

[uart] V1 Signoff

**Comment by rswarbrick** [comment]
Here are notes for uart verification signoff, following the
"light-weight signoff" procedure.

## Commits since Earlgrey-ES tapeout

Checked with version cfdbd6a386 (the head of `master` on 7/3/24).
Commits since the Earlgrey ES tapeout, found by
`git log Earlgrey-M2.5.2-RC0..HEAD --oneline hw/ip/uart`
(and skipping the reverted change):

- 0576a09be3 [uart] Correct timeout comment units
  + DV comment fix
- 0c580fad04 [uart] Further limit baud in uart_rx_oversample
  + DV constraint fix
- 8ea8f8f4f9 [uart,rtl,dv] Switch watermark interrupts to status type
  + RTL change: adjusting type for some interrupts
- 61a237e197 [util/reggen] reverse order of substruct generation
  + Reggen change
- de31bdf1c2 [reggen] Remove the devmode input
  + Reggen change
- eccaba59fb [doc] Updated Chromium EC URL
  + Doc change
- 5d25eb8536 [uart,sival] Add a feature list for the uart block
  + Sival change
- 1b16ca2122 [reggen] Add mubi support SWAccess that sets/clears a reg
  + Reggen change
- 59f8142826 [doc] Moved badges over to using hosted images
  + Doc change
- 3346485ce7 [doc] uart registers and interfaces now use CMDGEN
  + Doc change
- 7688e714e8 [reggen] Add initial support for version and cip_id hjson fields
  + Reggen change

I think that the only change in this list that could affect the
existing V2S status is

- 8ea8f8f4f9 [uart,rtl,dv] Switch watermark interrupts to status type

Looking through the V1/V2/V2S checklist, I don't believe that this
change could affect anything in the V1 or V2S lists. In the V2 list,
`SIM_ALL_TESTS_PASSING` is satisfied. `SIM_NIGHTLY_REGRESSION_V2` is
satisfied (93%):

![nightly-counts](https://github.com/lowRISC/opentitan/assets/104845/c3110991-bf4c-4505-8b9d-1ebecf623a53)

Similarly, `SIM_CODE_COVERAGE_V2` and `SIM_FUNCTIONAL_COVERAGE_V2` are
satisfied:

![coverage](https://github.com/lowRISC/opentitan/assets/104845/c70a858d-6bbf-47ea-b3a0-45a39a934933)

As such, I believe that the block's V2S status should remain unchanged.

## Issues closed since the Earlgrey-ES tapeout

There are an awful lot of issues closed in this time that match
"uart". To drop some unrelated sival/SW issues, I searched for:
```
is:issue is:closed closed:>2023-06-27 uart NOT chip-test NOT bazel
```

That search found the following items:

- #20469
  + SW issue
- #19564
  + OpenTitanTool
- #17819
  + OpenTitanTool
- #15785
  + spi_device issue (uart is only mentioned in passing)
- #15720
  + alert_handler issue (uart is only mentioned in passing)
- #15378
  + This issue tracks the RTL change that appeared in 8ea8f8f4f9 (and
    is discussed above)
- #13290
  + OpenTitanTool
- #8290
  + SW/usbdev issue that mentions uart in passing
- #4971
  + SCA tracking issue. Not really related to uart.
- #4334
  + Tooling issue
- #3097
  + DV question, not really related to uart.

The only change that might be related to V2S status is #15378, but
this is already discussed above.

## Currently open issues

Again, there are quite a few open issues that mention "uart". The
following search avoids some uninteresting sival/ottf/bazel/v1+v2 signoff results:
```
is:issue is:open uart NOT chip-test NOT opentitanlib NOT ottf NOT bazel NOT signoff
```

The results are:

- #21518
  + Questa tooling issue. Will not affect "V2S status"
- #19715
  + FPGA/SW usage question
- #19591
  + SW issue
- #18912
  + Not considered a bug
- #17348
  + Documentation issue and not related to uart.
- #16343
  + DV tooling issue not related to uart.
- #13873
  + SW issue that is unrelated to the uart block.
- #12989
  + 12989: Not related to uart block
- #10137
  + DV tooling issue, unrelated to uart block.
- #9134
  + Long-term system-level issue, unrelated to uart block
- #7826
  + Long-term system-level issue, unrelated to uart block
- #6517
  + DV tooling issue, unrelated to uart block.
- #6220
  + Long-term SW issue, unrelated to uart block
- #4887
  + Tooling issue
- #4743
  + Not related to uart block (usb FPGA tests)
- #4713
  + Not related to uart block (usbdev)
- #4689
  + DV tooling issue, unrelated to uart
- #3155
  + SW issue, unrelated to uart

None of these issues really affect the uart block at all, let alone
its verification status.

## Coverage report from <05/03/2024>

![prog-and-cover](https://github.com/lowRISC/opentitan/assets/104845/a4306ce0-b389-4b31-b708-983a0089b038)

Note that the V2 number isn't quite as bad as it looks. The 68.42%
number means that "68.42% of tests fail for every seed".

## Summary 

After a careful review, I'm convinced that the existing V2S status for
uart can stand.

Following the light-weight sign-off procedure document, I believe that
the next step is to "tag a TL (DV or DD)" to review the analysis above.

@msfschaffner: Would you mind looking through this? If you're happy
with the assessment, please close this issue.

**Comment by msfschaffner** [comment]
Overall this LGTM, and I apart from the interrupt alignments there do not seem to be major changes here.
However, regarding the pass rates of individual tests in the nightly, we do have several that are below 90%.
If we want to keep the existing V2S status, I think we should take another look and patch the DV accordingly.

**Comment by msfschaffner** [comment]
From last Tuesday:
![image](https://github.com/lowRISC/opentitan/assets/41358501/c5de4286-cf96-4c91-9857-8d72c573f1d1)

**Comment by rswarbrick** [comment]
Cool. @elliotb-lowrisc is actually working on these tests at the moment (started in the last couple of days). I'm optimistic that the pass rates are going to get above 90% very soon.

(Except possibly `stress_all_with_reset`: why on earth did we put this in as V2 in the first place?!)

**Comment by rswarbrick** [comment]
I think we're probably good to sign off now. Tuesday's nightly result are well over 90%, except for `uart_stress_all_with_rand_reset`.

**Comment by andreaskurth** [comment]
Keeping V2S for uart SGTM based on the analysis above (thanks @rswarbrick :+1: and the latest nightly regression results, which show 100% pass rate for all tests except `uart_stress_all_with_rand_reset`, which I agree with Rupert that this should be in V3. Could you please move it to V3 in a follow-up PR?

**Comment by rswarbrick** [comment]
Thanks! The V3 move will be implemented by #22220.

**Comment by vogelpi** [comment]
Note that UART has effectively been signed off at V2S. I've now adjusted the issue title and first comment accordingly.
